smarts/core/agent.py

```diff
@@ -146,7 +146,34 @@ def __post_init__(self):
     def replace(self, **kwargs) -> "AgentSpec":
         """Return a copy of this AgentSpec with the given fields updated."""
 
-        return replace(self, **kwargs)
+        replacements = [
+            ("policy_builder", "agent_builder"),
+            ("policy_params", "agent_params"),
+            ("perform_self_test", None),
+        ]
+
+        assert None not in kwargs, f"Error: replace contains invalid key: ({None})"
```

Comment:
I guess `f" "` is not needed as there is no string formatting. Alternatively, we can improve the error message as:

```suggestion
        assert None not in kwargs, f"Error: kwargs input to replace() function contains invalid key `None`: {kwargs}"

```

Reply:
This seems like a reasonable change.